gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/grpc.py.j2

```diff
@@ -133,8 +133,8 @@ class {{ service.name }}GrpcTransport({{ service.name }}Transport):
             warnings.warn("client_cert_source is deprecated", DeprecationWarning)
 
         if channel:
-            # Ignore credentials if a channel was passed.
-            credentials = False
+            # Ignore credentials if a channel was passed without credentials.
+            credentials = False if not credentials else credentials
```

Comment:
This wasn't a part of the design so adding some clarification. Validation step (for universe domain) requires us to access the universe domain from the credentials, hence we can no longer ignore credentials if a channel is explicitly passed in (mainly for testing purposes).
i.e. the condition:
```python3
if self.universe_domain != self.transport._credentials.universe_domain
```
fails without this change.

Reply:
Would this be a breaking change? Before, if the user was passing a channel with credentials, we were clearing the credentials. Now we're not. So that will affect the behavior of this constructor, it would seem.

One idea for not changing the code is that when we try to access the universe domain from the credentials, we first check to see whether the credentials are set, and if they're not, consider the universe domain to be the empty string or `None`. Would this help the testing scenario?

Reply:
Addressed!